mediacheck.md

```diff
@@ -0,0 +1,151 @@
+# libmediacheck
+
+## About
+
+`libmediacheck` is a library for verifying SUSE installaton media. It is used by [checkmedia](README.md) and
+[linuxrc](https://github.com/openSUSE/linuxrc).
+
+The library exports two groups of functions:
+
+1. functions for verifying SUSE media
+2. functions for digest calculations
+
+The second group is there for convenience to be used by `linuxrc`.
+
+## API functions for media verification
+
+Have a look at [checkmedia.c](checkmedia.c) for a simple usage example.
+
+### Create new mediacheck object
+
+```
+mediacheck_t *mediacheck_init(char *file_name, mediacheck_progress_t progress);
+
+typedef int (* mediacheck_progress_t)(unsigned percent);
+```
+
+- `file_name` is the name of the image file (or device) to check.
+
+- `progress` is a function that will be called at regular intervals to
+indicate the verification progress. The function will be typically called
+once for each additional percent progressed. The first call is guaranteed to
+be for 0 %, the last for 100 %.
+
+`progress` may return 0 or 1. 1 indicates that the check should be aborted.
```

Comment:
+1 for documenting it. But it would be nice to also document it in header file as part of API.